exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -130,13 +131,20 @@ func (amp *AzureMachinePool) ValidateNetwork() error {
 	return nil
 }
 
+// ValidateOSDisk of an AzureMachinePool.
+func (amp *AzureMachinePool) ValidateOSDisk() error {
+	if errs := infrav1.ValidateOSDisk(amp.Spec.Template.OSDisk, field.NewPath("osDisk")); len(errs) > 0 {
```

Comment:
Reusing the same validation meant I had to add the defaults for OSType and CachingType. Before there was no validation done for OSDisks and in general not a lot of things get validated for AzureMachinePools (quite the contrast to AzureMachines). Not sure if there should be a refactoring to bring this more up to par.

Reply:
Good catch, I think it'll be good to consider some refactoring if the fields aren't being validated in the same way here.

Reply:
yeah definitively. I started with #4927 a few days ago and there the same issue will appear. A refactoring is needed to make the validation uniform.